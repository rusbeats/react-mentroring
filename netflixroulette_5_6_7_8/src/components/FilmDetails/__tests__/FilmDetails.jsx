import React from 'react';
import { shallow } from 'enzyme';
import { FilmDetails } from '../index';

const PROPS = {
    match: {
        params: 1
    },
    film: {
        id: 1,
        genres: ['genre1', 'genre2', 'genre3'],
        runtime: 100
    },
    film2: {
        id: 2,
        genres: ['genre4'],
        runtime: 100
    },
    notFound: true
};

describe('FilmDetails', () => {
    it('should render correctly (snapshot)', () => {
        const component = shallow(<FilmDetails match={PROPS.match} filmFromState={PROPS.film} />);
        expect(component).toMatchSnapshot();
    });

    it('should render correctly if no film found (snapshot)', () => {
        const component = shallow(
            <FilmDetails match={PROPS.match} filmFromState={PROPS.film2} notFound={PROPS.notFound} />
        );
        expect(component).toMatchSnapshot();
    });
});
