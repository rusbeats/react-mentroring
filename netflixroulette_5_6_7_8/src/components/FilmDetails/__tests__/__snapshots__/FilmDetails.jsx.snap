// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmDetails should render correctly (snapshot) 1`] = `
<div>
  <FilmDetailsstyles__StyledDetails>
    <FilmDetailsstyles__FilmDetailsHeaderWrapper>
      <FilmDetailsstyles__FilmDetailsHeaderText>
        netflixRoulette
      </FilmDetailsstyles__FilmDetailsHeaderText>
      <Link
        href="/"
      >
        <Linkstyles__StyledLink>
          <FilmDetailsstyles__FilmDetailsSearchButton>
            search
          </FilmDetailsstyles__FilmDetailsSearchButton>
        </Linkstyles__StyledLink>
      </Link>
    </FilmDetailsstyles__FilmDetailsHeaderWrapper>
    <FilmDetailsstyles__FilmDetailsWrapper>
      <Imagestyles__Image />
      <FilmDetailsstyles__FilmInfo>
        <FilmDetailsstyles__FilmTitleAndRating>
          <FilmDetailsstyles__FilmTitle />
          <FilmDetailsstyles__FilmRating />
        </FilmDetailsstyles__FilmTitleAndRating>
        <FilmDetailsstyles__FilmYearAndDuration>
          <FilmDetailsstyles__FilmYear>
            NaN
          </FilmDetailsstyles__FilmYear>
          <div>
            100
             
            filmsDetails.minuteAbbr
          </div>
        </FilmDetailsstyles__FilmYearAndDuration>
        <FilmDetailsstyles__FilmOverview />
      </FilmDetailsstyles__FilmInfo>
    </FilmDetailsstyles__FilmDetailsWrapper>
  </FilmDetailsstyles__StyledDetails>
  <FilmDetailsstyles__FilmsByGenre>
    filmsDetails.filmsByGenre
  </FilmDetailsstyles__FilmsByGenre>
</div>
`;

exports[`FilmDetails should render correctly if no film found (snapshot) 1`] = `<NotFound />`;
